src/statefunctions/router.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include <map>
+#include <memory>
+
+#include "types/statefunction.h"
+
+namespace mahjong {
+class Router {
+ public:
+  static Router& Instance() {
+    static Router router;
+    return router;
+  }
+
+  StateFunctionFunc Route(const StateFunctionType& type);
+  bool RegisterRoute(StateFunctionFunc func, const StateFunctionType& type);
+
+ private:
+  std::map<StateFunctionType, StateFunctionFunc> routes_;
```

Comment:
is the underscore suffix standard for private properties?

Reply:
Yep basically an alternative to m_routes. And more specifically it's for all non-static member variables, but non-private variables are a bad idea so basically yes.